src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prev = initialValue;
+    let start = 0;
+
+    if (arguments.length === 1) {
```

Comment:
В этом месте мы проверяем было ли передано initialValue в функцию. Если нет, то значение = undefined;
Более логично проверять именно это.
```suggestion
    if (prev === undefined) {
```

Reply:
Yep, it would be more semantical, but this variant is ok as well